PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -25,6 +26,33 @@
 SEED_LEN = 64
 
 
+def xor_byte_strings(b1: bytes, b2: bytes) -> bytes:
+    """Function for XOR'ing two objects of byte type
+    Reference implementation: https://en.wikipedia.org/wiki/XOR_cipher
+    :rtype: bytes
+    :return b1 XOR b2
+    """
+    return bytes([x ^ y for x, y in zip(b1, b2)])
+
+
+def pbkdf2_sha512(passw: bytes, salt: bytes, rounds: int) -> bytes:
+    """Password Based Key Derivation Function
+    https://en.wikipedia.org/wiki/PBKDF2
+    Uses HMAC-SHA512, derived key length is 512 bit
+    :rtype: bytes
+    :return: key derived by PBKDF2 algorithm
+    """
+    # The first iteration of PRF uses Password as the PRF key
+    # and Salt concatenated with i encoded as a big-endian
+    # 32-bit integer as the input.
+    U = hmac.new(passw, salt + (1).to_bytes(4, byteorder='big'), digestmod=sha512).digest()
+    F = U  # F is the xor (^) of c iterations of chained PRFs
+    for i in range(1, rounds):
```

Comment:
> he iteration count is set to 2048
> https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#from-mnemonic-to-seed

`range(1, 2048)` is 2047 iterations. `len(range(1, 2048)) == 2047`

Reply:
First iteration is  U = hmac.new(passw, salt + (1).to_bytes(4, byteorder='big'), digestmod=sha512).digest(), it is computed manually, [U1 = PRF(Password, Salt + INT_32_BE(i))](https://en.wikipedia.org/wiki/PBKDF2#Key_derivation_process)

Reply:
I don't know if we have to generalize the implementation to the arbitrary key length, I've just hardcoded that.

Reply:
Didn't recognize that, thanks for explaining! :)

I'm fine with `pbkdf2_sha512` with fixed key length. Outside of this project we would use functions for this from standard library, no need to implement general solution for that. :)